--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -219,6 +219,11 @@ static void option_instat_callback(struct urb *urb);
 #define WETELECOM_PRODUCT_6802			0x6802
 #define WETELECOM_PRODUCT_WMD300		0x6803
 
+/* ThinkWill Products */
+#define THINKWILL_LTE_VENDOR_ID			0x19d2
+#define THINKWILL_PRODUCT_ML7820_ECM		0x0199	/* ECM mode */
+#define THINKWILL_PRODUCT_ML7820_NDIS		0x0532	/* NDIS mode */
+#define THINKWILL_PRODUCT_ML7820_DL		0x0256	/* Download mode */
 
 /* Device flags */
 
@@ -1091,6 +1096,9 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(ALINK_VENDOR_ID, ALINK_PRODUCT_3GU) },
 	{ USB_DEVICE(ALINK_VENDOR_ID, SIMCOM_PRODUCT_SIM7100E),
 	  .driver_info = RSVD(5) | RSVD(6) },
+	{ USB_DEVICE(THINKWILL_LTE_VENDOR_ID, THINKWILL_PRODUCT_ML7820_ECM) },
+	{ USB_DEVICE(THINKWILL_LTE_VENDOR_ID, THINKWILL_PRODUCT_ML7820_NDIS) },
+	{ USB_DEVICE(THINKWILL_LTE_VENDOR_ID, THINKWILL_PRODUCT_ML7820_DL) },
 	{ USB_DEVICE(ALCATEL_VENDOR_ID, ALCATEL_PRODUCT_X060S_X200),
 	  .driver_info = NCTRL(0) | NCTRL(1) | RSVD(4) },
 	{ USB_DEVICE(ALCATEL_VENDOR_ID, ALCATEL_PRODUCT_X220_X500D),
