--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -1081,6 +1081,9 @@ static const struct usb_device_id option_ids[] = {
 	{ USB_DEVICE(ALINK_VENDOR_ID, SIMCOM_PRODUCT_SIM7100E),
 	  .driver_info = RSVD(5) | RSVD(6) },
 	{ USB_DEVICE_INTERFACE_CLASS(0x1e0e, 0x9003, 0xff) },	/* Simcom SIM7500/SIM7600 MBIM mode */
+	/* ThinkWill Devices */
+	{ USB_DEVICE(0x19d2, 0x0199) },
+	{ USB_DEVICE(0x19d2, 0x0532) },
+	{ USB_DEVICE(0x19d2, 0x0256) },
